decode_morse.rb

```diff
@@ -0,0 +1,50 @@
+MORSE = {
+  '.-' => 'A',
+  '-...' => 'B',
+  '-.-.' => 'C',
+  '-..' => 'D',
+  '.' => 'E',
+  '..-.' => 'F',
+  '--.' => 'G',
+  '....' => 'H',
+  '..' => 'I',
+  '.---' => 'J',
+  '-.-' => 'K',
+  '.-..' => 'L',
+  '--' => 'M',
+  '-.' => 'N',
+  '---' => 'O',
+  '.--.' => 'P',
+  '--.-' => 'Q',
+  '.-.' => 'R',
+  '...' => 'S',
+  '-' => 'T',
+  '..-' => 'U',
+  '...-' => 'V',
+  '.--' => 'W',
+  '-..-' => 'X',
+  '-.--' => 'Y',
+  '--..' => 'Z'
+}.freeze
+
+def decode_char(str)
+  MORSE.each do |key, value|
+    return value.to_s if key == str
+  end
+end
+
+def decode_word(str)
+  message = ''
+  str.split.each do |i|
+    message += decode_char(i).to_s
+  end
+  message
+end
+
+def decode(str)
+  message = []
+  str.split('  ').each do |i|
+    message.push(decode_word(i).to_s)
+  end
```

Comment:
- [ ] the words in the message are split by 3 spaces, however in this method are only 2 spaces, because of this the below piece of code:
```
decode(".-   -... --- -..-   ..-. ..- .-.. .-..    --- ..-.    .-. ..- -... .. . ...")
```
translates to:
`"A BOX FULL  OF  RUBIES"`
As you can see there are too many spaces between the words "FULL", "OF" and "RUBIES"